src/problem.ts

```diff
@@ -1,62 +1,77 @@
-    public http?: httpObject
-    [key: string]: any;
-
-
-    constructor(problem: ProblemInterface, customKeys?: string[]) {
-        super(problem.detail || problem.title);
-        this.http = problem.http
-        this.type = problem.type
-        this.title = problem.title;
-        this.detail = problem.detail;
-        this.instance = problem.instance;
-        this.stack = problem.stack;
-        customKeys?.map((customKey) => {
-            this[customKey] = problem[customKey];
-        })
-    }
-
-   copy(problem: ProblemInterface, mode: COPY_MODE, props: string[]): Problem {
-        switch (mode) {
-
-            case COPY_MODE.LEAVE_PROPS:
-                return new Problem(problem, props);
-
-            case COPY_MODE.SKIP_PROPS:
-            default:
-                let keysToBeCopied: string[] = [];
-                for (let key in problem) {
-                    if (props.includes(key))
-                        continue;
-                    keysToBeCopied.push(key)
-                }
-                return new Problem(problem, keysToBeCopied)
-        }
-    };
+  public type: string;
+  public title: string;
+  public instance?: string;
+  public detail?: string;
+  public http?: httpObject;
+  [key: string]: any;
 
-   toJSON(problem: Problem, includeStack = false): ProblemInterface {
+  constructor(problem: ProblemInterface, customKeys?: string[]) {
+    super(problem.detail || problem.title);
+    this.http = problem.http;
+    this.type = problem.type;
+    this.title = problem.title;
+    this.detail = problem.detail;
+    this.instance = problem.instance;
+    this.stack = problem.stack;
+    customKeys?.map((customKey) => {
+      this[customKey] = problem[customKey];
+    });
+  }
 
-        const { name, message, stack, ...rest } = problem;
+  copy(problem: ProblemInterface, mode: COPY_MODE, props: string[]): Problem {
+    switch (mode) {
+      case COPY_MODE.LEAVE_PROPS:
+        return new Problem(problem, props);
 
-        const jsonObject = {
-            ...rest
+      case COPY_MODE.SKIP_PROPS:
+      default:
+        let keysToBeCopied: string[] = [];
+        for (let key in problem) {
+          if (props.includes(key)) continue;
+          keysToBeCopied.push(key);
         }
+        return new Problem(
+          { type: problem.type, title: problem.title },
+          keysToBeCopied
+        );
+    }
+  }
 
-        if (includeStack)
-            jsonObject.stack = stack;
-
-        return jsonObject;
+  toJSON({ includeStack = false, stringify = false }: ToJsonParamType) {
+    const thisContext = this;
+    const { stack, ...rest } = thisContext;
 
+    if (includeStack) {
+      if (stringify) return JSON.stringify(thisContext);
+      return {
+        ...thisContext,
+        stack: this.stack,
+      };
     }
-};
\ No newline at end of file
+
+    if (stringify) return JSON.stringify({ ...rest });
+    return { ...rest };
+  }
+
+  isOfType = (type: string) => this.type === type;
```

Comment:
Change that to the function, because atm you treat is a property:

```suggestion
  isOfType(type: string) {
    return this.type === type;
  }
```